hw/ip/prim/dv/prim_ascon/ascon_model_dpi/ascon_model_dpi.c

```diff
@@ -0,0 +1,74 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ascon_model_dpi.h"
+
+#include <assert.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "svdpi.h"
+#include "vendor/ascon_ascon-c/ascon128/round.h"
+
+void c_dpi_ascon_round(const svBitVecVal *data_i, svBit *round_i,
+                       svBitVecVal *data_o) {
+  uint8_t round;
+
+  // get input data from simulator
+  ascon_state_t *state = ascon_data_get(data_i);
+  round = (uint8_t)*round_i;
+
+  ROUND(state, round);
+
+  ascon_data_put(data_o, state);
+
+  return;
+}
+
+ascon_state_t *ascon_data_get(const svBitVecVal *data_i) {
+  ascon_state_t *data;
+  svBitVecVal value;
+
+  // alloc data buffer
+  data = (ascon_state_t *)malloc(sizeof(ascon_state_t));
+  assert(data);
+
+  // get data from simulator, convert from 2D to 1D
+  printf("Getter:\n");
+  for (int i = 0; i < 5; i++) {
+    value = data_i[2 * i];
+    data->x[i] = (uint64_t)value;
```

Comment:
Is there any reason not to have the cast (also above)? So far I encountered many bugs due to missing casts, but none due to an unneeded one (at least not for non strong typed languages). Especially, as it is needed for the upper word, it think it is more readable if both cases are structured the same way.

However, as I'm really not very experienced with C, I'm happy to be convinced.

Reply:
Definitely no strong reason. I think I'd normally be inclined to avoid casts unless we're actually squashing something down or (more confusingly) messing around with signedness. But adding one here definitely doesn't cause any harm. Let's not worry about it :-)